src/Impostor.Server/Http/ListingManager.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections.Generic;
+using System.Linq;
+using Impostor.Api.Games;
+using Impostor.Api.Games.Managers;
+using Impostor.Api.Http;
+using Impostor.Api.Innersloth;
+using Microsoft.AspNetCore.Http;
+
+namespace Impostor.Server.Http;
+
+/// <summary>
+/// Perform game listing filtering.
+/// </summary>
+public sealed class ListingManager
+{
+    private readonly IGameManager _gameManager;
+    private readonly IEnumerable<IListingFilter> _listingFilters;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ListingManager"/> class.
+    /// </summary>
+    /// <param name="gameManager">GameManager containing a list of games.</param>
+    /// <param name="listingFilters">Filters used when finding game listings.</param>
+    public ListingManager(IGameManager gameManager, IEnumerable<IListingFilter> listingFilters)
+    {
+        _gameManager = gameManager;
+        _listingFilters = listingFilters;
+    }
+
+    /// <summary>
+    /// Find listings that match the requested settings.
+    /// </summary>
+    /// <param name="ctx">The context of this http request.</param>
+    /// <param name="map">The selected maps.</param>
+    /// <param name="impostorCount">The amount of impostors. 0 is any.</param>
+    /// <param name="language">Chat language of the game.</param>
+    /// <param name="maxListings">Maximum amount of games to return.</param>
+    /// <returns>Listings that match the required criteria.</returns>
+    public IEnumerable<IGame> FindListings(HttpContext ctx, int map, int impostorCount, GameKeywords language, int maxListings = 10)
```

Comment:
This method is a dupe of GameManager.FindListings (although we improved it afterwards)
Do we want to merge these functions back together again or should we remove game list requesting over udp?

Merging may be difficult because of the ListingFilter getting an HttpContext as an argument, which was the whole reason for the split

Reply:
I was going to ask about this but forgot, imo we should just remove udp matchmaking to make the code cleaner

Reply:
There is not actually much we can remove, joining via code or creating a new game still happens through udp mm. Only the gamelist stuff can actually be removed